cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,138 @@
+package memory
```

Comment:
I agree that having a whole bunch of packages named "memory" is weird.

I think it's important to separate the heavy-duty implementation into a separate package (especially given that most of our production implementations will be closed-source). I find it's good practice to have the simple, fake implementation live in a separate package (to make sure that implementations don't require unexported features of the interface).